disnake/state.py

```diff
@@ -782,14 +783,22 @@ def parse_message_create(self, data: gateway.MessageCreateEvent) -> None:
         if self._messages is not None:
             self._messages.append(message)
         # we ensure that the channel is a type that implements last_message_id
-        if channel and channel.__class__ in (TextChannel, Thread, VoiceChannel):
-            channel.last_message_id = message.id  # type: ignore
+        if channel:
+            if channel.__class__ in (TextChannel, Thread, VoiceChannel):
+                channel.last_message_id = message.id  # type: ignore
+            if channel.__class__ is Thread and message.type is not MessageType.thread_starter_message:
+                channel.total_message_sent += 1
```

Comment:
missing a `# type: ignore`

Reply:
> missing a `# type: ignore`

so in conclusion what should I use? `.__class__`, `type()`, `isinstance()`, `typing.cast()` or just `# type: ignore`?